virtcontainers/kata_agent.go

```diff
@@ -1134,6 +1138,30 @@ func (k *kataAgent) handleEphemeralStorage(mounts []specs.Mount) []*grpc.Storage
 	return epheStorages
 }
 
+// handleLocalStorage handles local storage within the VM
+// by creating a directory in the VM from the source of the mount point
+func (k *kataAgent) handleLocalStorage(mounts []specs.Mount, sandboxID string) []*grpc.Storage {
+	var localStorages []*grpc.Storage
+	for idx, mnt := range mounts {
+		if mnt.Type == kataLocalDevType {
+			// Set the mount source path to a the desired directory point in the VM.
+			// In this case it is located in the sandbox directory.
+			mounts[idx].Source = filepath.Join(kataGuestSharedDir, sandboxID, kataLocalDevType, filepath.Base(mnt.Source))
```

Comment:
What if two empty-dirs have the same Base? Could you make sure that this path is unique.

Reply:
It isn't possible for two emptyDirs to have the same base, see example pod spec:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  containers:
    - name: test
      image: ubuntu
      args:
        - sleep
        - "3600"
  volumes:
    - name: test123
      emptyDir:
        medium: ""
    - name: test123
      emptyDir:
        medium: ""
```
```
$ k create -f test.yaml
The Pod "test" is invalid: spec.volumes[1].name: Duplicate value: "test123"
```

Should I still make it unique?

Reply:
`kataGuestSharedDir` is actually tmpfs in the guest. Is it desirable to use memory for k8s empty dir? The other options are to use guest rootfs or container rootfs.

Reply:
It is **NOT** desirable to use memory for k8s emptyDir (unless it is of `medium: Memory`) and in this case it is **NOT** on a tmpfs. The full path for the emptyDir directory will be: `/run/kata-containers/shared/containers/<sandbox id>/local/<emptyDir name>`, which is on whatever storage the sandbox container (i.e. the pause container is using).

See comment by @mcastelino:
> Using the pause container for holding on to the ephemeral volumes makes a lot of sense. It keeps the individual container behaviour unchanged. This also allows the volume to grow to the limit afforded by the graph driver.

Reply:
@awprice OK, I see. You are relying on the fact that the sandbox id is the same as the first container id. We don't guarantee that but it is also the current fact. Please add some comments above the lines to highlight it so that we don't forget to fix it in case we need to change it.